core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -330,8 +338,16 @@ private Object handleSecondPhaseCommand(TxInvocationContext ctx, TransactionBoun
    private Collection<Address> getCommitNodes(TxInvocationContext ctx, TopologyAffectedCommand command) {
       LocalTransaction localTx = (LocalTransaction) ctx.getCacheTransaction();
       LocalizedCacheTopology cacheTopology = CacheTopologyUtil.checkTopology(command, getCacheTopology());
-      Collection<Address> affectedNodes =
-            isReplicated ? null : cacheTopology.getWriteOwners(ctx.getAffectedKeys());
+      // Utilize all the keys to identify the nodes.
+      // Otherwise, a local transaction transferred remotely by other node during state transfer might not be completed.
```

Comment:
The node transfers transactions based on the locked keys in the requested segments during state transfer. The TX owner doesn't transfer the TX if it doesn't have a lock with the local transaction. However, the donor might have acquired a remote lock, which causes the remote node to transfer the remote transaction without the original TX owner's knowledge. 

When the TX owner sends the TX complete notification, it doesn't know that the new member has received the TX and never sends the notification to it.